src/core/SparseMatrix.cpp

```diff
@@ -68,6 +68,20 @@ SparseMatrix::SetZeros(){
     fArmaMat = arma::sp_mat(fNCols, fNRows);
 }
 
+void
+SparseMatrix::SetIdentity(){
+    if(fNRows && fNCols==fNRows )
+        throw DimensionError(Formatter()<<
+                "SparseMatrix:: Can't set identity as matrix is not square. (rows,cols) : ("<<
+                fNRows<<","<<fNCols<<")"  
```

Comment:
Don't be sorry, I don't understand if ```fNRows = 0```, ```fNCols = 1``` the error gets thrown. What am i missing ?